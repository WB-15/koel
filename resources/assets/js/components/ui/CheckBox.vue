<template>
  <span>
    <input :checked="checked" type="checkbox" v-bind="$attrs" @input="onInput">
    <icon v-if="checked" :icon="faCheck" />
  </span>
</template>

<script lang="ts" setup>
import { faCheck } from '@fortawesome/free-solid-svg-icons'
import { ref } from 'vue'

const props = withDefaults(defineProps<{ modelValue?: any }>(), {
  modelValue: false
})

const checked = ref(props.modelValue)

const emit = defineEmits<{ (e: 'update:modelValue', value: boolean): void }>()

const onInput = (event: Event) => {
  checked.value = (event.target as HTMLInputElement).checked
  emit('update:modelValue', checked.value)
}
</script>

<style scoped>
span {
  position: relative;
}

svg {
  color: var(--color-highlight);
  position: absolute;
  top: 1px;
  left: 2px;
}
</style>
