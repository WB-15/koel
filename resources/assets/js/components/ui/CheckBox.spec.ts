import { expect, it } from 'vitest'
import UnitTestCase from '@/__tests__/UnitTestCase'
import CheckBox from './CheckBox.vue'
import { fireEvent } from '@testing-library/vue'

new class extends UnitTestCase {
  protected test () {
    it('renders unchecked state', () => expect(this.render(CheckBox).html()).toMatchSnapshot())

    it('renders checked state', () => expect(this.render(CheckBox, {
      props: {
        modelValue: true
      }
    }).html()).toMatchSnapshot())

    it('emits the input event', async () => {
      const { getByRole, emitted } = this.render(CheckBox)

      await fireEvent.input(getByRole('checkbox'))

      expect(emitted()['update:modelValue']).toBeTruthy()
    })
  }
}
