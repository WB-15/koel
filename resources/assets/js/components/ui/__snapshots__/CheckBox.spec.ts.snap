// Vitest Snapshot v1

exports[`renders checked state 1`] = `<span data-v-b5259680=""><input data-v-b5259680="" type="checkbox"><br data-v-b5259680="" data-testid="icon" icon="[object Object]"></span>`;

exports[`renders unchecked state 1`] = `<span data-v-b5259680=""><input data-v-b5259680="" type="checkbox"><!--v-if--></span>`;
